src/components/Modal/BaseModal.js

```diff
@@ -73,6 +74,9 @@ class BaseModal extends PureComponent {
             this.props.onModalHide();
         }
         Modal.onModalDidClose();
+        if (!this.props.fullscreen) {
```

Comment:
what's this condition for?

Reply:
Hey @ntdiary, I know this thread is kinda old, but would appreciate your input:
Why was it necessary to put the `ComposerFocusManager.setReadyToFocus()` logic into both `onDismiss` and `onModalHide` (which calls `hideModal`)?
Was there any drawback to just calling it in `hideModal` regardless of the `fullscreen` condition? This would mean removing the `ReactNativeModal.onDismiss` prop below.
I'm asking because we've recently discovered another case when the `onDismiss` is not called, this time for a full-screen Attachments modal.

Reply:
Thanks @ntdiary. Do you have any particular `Modal` scenario in mind that might break?
I tried with only the `hideModal`: emoji picker, attachments modal, the side drawer – everything focuses the composer correctly

Reply:
Cool, would your code cover the modal being closed on clicking the browser's "Back" button?